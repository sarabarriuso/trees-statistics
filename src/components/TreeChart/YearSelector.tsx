import Slider from '@mui/material/Slider';
interface IMonthSelectorProps {
  description: string;
  defaultYear: number;
  minYear?: number;
  yearSelectedCallback: (value: number) => void;
}
const MonthSelector: React.FC<IMonthSelectorProps> = ({
  description,
  defaultYear,
  minYear,
  yearSelectedCallback,
}) => {
  const DEFAULT_CLASSNAME = 'year-selector';

  const handleChange = (
    _event: Event | React.SyntheticEvent<Element, Event>,
    value: number | number[],
  ) => {
    yearSelectedCallback(value as number);
  };

  const marks = [];
  for (let i = 2019; i <= new Date().getFullYear(); i++) {
    marks.push({ value: i, label: i.toString() });
  }

  return (
    <div className={DEFAULT_CLASSNAME}>
      <p>{description}</p>
      <Slider
        aria-label='Day'
        defaultValue={defaultYear}
        track={false}
        valueLabelDisplay='auto'
        step={1}
        marks={marks}
        min={minYear || 2019}
        max={new Date().getFullYear()}
        onChangeCommitted={handleChange}
      />
    </div>
  );
};

export default MonthSelector;
