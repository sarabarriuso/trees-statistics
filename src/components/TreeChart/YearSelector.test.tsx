import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import YearSelector from './YearSelector';
import { Slider } from '@mui/material';

export default describe('Year Selector Index Tests', () => {
  it('Renders successfully ', () => {
    // Arrange
    // Act
    const tree = shallow(
      <YearSelector
        description='Hi I am a description'
        defaultYear={1}
        yearSelectedCallback={() => {
          return;
        }}
      />,
    );

    //Assert
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('Changing the year will trigger the yearSelectedCallback', () => {
    // Arrange
    let callbackCalled = false;

    // Act
    const tree = shallow(
      <YearSelector
        description='Hi I am a description'
        defaultYear={1}
        yearSelectedCallback={() => {
          callbackCalled = true;
        }}
      />,
    );

    const slider = tree.find(Slider);
    const sliderCallback = slider.props().onChangeCommitted;
    if (sliderCallback !== undefined) {
      sliderCallback(new Event('click', undefined), 2020);

      //Assert
      expect(callbackCalled).toBeTruthy();
    }
  });
});
