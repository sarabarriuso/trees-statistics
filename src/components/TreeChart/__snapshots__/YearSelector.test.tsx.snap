// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Year Selector Index Tests Renders successfully  1`] = `
<div
  className="year-selector"
>
  <p>
    Hi I am a description
  </p>
  <ForwardRef(Slider)
    aria-label="Day"
    defaultValue={1}
    marks={
      Array [
        Object {
          "label": "2019",
          "value": 2019,
        },
        Object {
          "label": "2020",
          "value": 2020,
        },
        Object {
          "label": "2021",
          "value": 2021,
        },
        Object {
          "label": "2022",
          "value": 2022,
        },
      ]
    }
    max={2022}
    min={2019}
    onChangeCommitted={[Function]}
    step={1}
    track={false}
    valueLabelDisplay="auto"
  />
</div>
`;
